data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/Main.java

```diff
@@ -110,34 +111,19 @@ public static void main(final String[] args) throws IOException {
       producerConfigs.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, CloudEventSerializer.class);
       producerConfigs.put(ProducerConfig.INTERCEPTOR_CLASSES_CONFIG, PartitionKeyExtensionInterceptor.class.getName());
 
-      final Function<Vertx, RequestMapper> handlerFactory = v -> new RequestMapper(
-        v,
-        AuthProvider.kubernetes(),
-        producerConfigs,
-        new CloudEventRequestToRecordMapper(),
-        properties -> KafkaProducer.create(v, properties),
-        badRequestCounter,
-        produceEventsCounter
-      );
-
       final var httpServerOptions = new HttpServerOptions(
         Configurations.getPropertiesAsJson(env.getHttpServerConfigFilePath())
       );
       httpServerOptions.setPort(env.getIngressPort());
       httpServerOptions.setTracingPolicy(TracingPolicy.PROPAGATE);
 
-      final var verticle = new ReceiverVerticle(
-        httpServerOptions,
-        handlerFactory,
-        h -> new SimpleProbeHandlerDecorator(
-          env.getLivenessProbePath(),
-          env.getReadinessProbePath(),
-          h
-        )
-      );
+      final Supplier<Verticle> verticle = new ReceiverVerticleSupplier(env, producerConfigs, badRequestCounter, produceEventsCounter, httpServerOptions);
```

Comment:
Can we just inline that lambda more than creating a new class?

Reply:
No, I prefer to keep it in a separate class simpler to read and unit testable.